### PR TITLE
fix(nextcloud)!: Throw DynamiteStatusCodeExceptions for invalid WebDAV status codes

### PR DESCRIPTION
**Pull request by provokateurin**
Fixes https://github.com/nextcloud/neon/issues/2307

### CODE REVIEW

File: packages/nextcloud/lib/src/webdav/webdav_client.dart [hunk 1]
```diff
@@ -355,40 +390,33 @@ class WebDavClient {
...
-            onDone: () {
-              onProgress?.call(1);
-              controller.close();
-            },
-          );
-        },
-        // ignore: avoid_types_on_closure_parameters
-        onError: (Object error) {
-          controller.addError(error);
-        },
```
Comment (Leptopoda) [inline packages/nextcloud/lib/src/webdav/webdav_client.dart:390]: Why did you remove the on error callback?

File: packages/nextcloud/test/webdav/webdav_test.dart
Comment (Leptopoda) [inline packages/nextcloud/test/webdav/webdav_test.dart:1]: Please add tests for the invalid status code (probably requires a mocked http client)

**Review by Leptopoda — CHANGES_REQUESTED** [review]
I'm not so sure about the first commit.
Maybe we should wait with any refactors in that regard until we've decided on how we tackle other webdav based clients.

File: packages/nextcloud/lib/src/webdav/webdav_client.dart (hunk 1 above)
Comment (provokateurin) [reply packages/nextcloud/lib/src/webdav/webdav_client.dart:390]: This was needed so the error would appear in the stream. The function is now async and the error will happen on the await and not in the stream. You can check the test for this, it still works.

**Comment by provokateurin** [comment]
>  Maybe we should wait with any refactors in that regard until we've decided on how we tackle other webdav based clients.

I think any refactor to support CalDAV and CardDAV will need this even more as the clients get even more different from the generated ones by extending the base DAV client in some form (e.g. polymorphism or some other way)

**Review by Leptopoda — APPROVED** [review]
